lua/telekasten.lua

```diff
@@ -1446,10 +1456,12 @@ local function RenameNote()
     if M.Cfg.rename_update_links == true then
         -- Only look for the first part of the link, so we do not touch to #heading or #^paragraph
         -- Should use regex instead to ensure it is a proper link
-        -- Should also account for other types of links (?)
         local oldlink = "\\[\\[" .. oldname
         local newlink = "\\[\\[" .. newname
 
+        -- Save all open buffers before looking for links to replace
```

Comment:
I know I suggested it. But I am still debating myself here: should we 

a) warn the user that we're going to write all buffers now?
b) put in the documentation that __ALL__ buffers are going to be saved

I fear that some users will make changes in completely unrelated buffers they don't want to be saved (yet). Especially non-telekasten related buffers.

So, as tedious as that may seem, my current opinion is:
- put in the documentation that this command will save all **telekasten** buffers (no others)
- iterate over all open buffers, find out their filetypes and only save those which match "telekasten"

If some people use other filetypes, like rmarkdown, it is their responsibility to save their files before closing - in order to expect potential link renames to happen. (Note: if someone uses `.rmd` as telekasten extension, this will not count as using another filetype, as telekasten always sets its own filetype for files opened via telekasten). So I actually doubt someone uses telekasten "functions" in other filetypes - but one can never be certain.

Reply:
Yes, this was basically a quickfix to see if it worked as expected. I do not like forcing users to save unrelated stuff either.

I would prefer it to save only telekasten buffers. I can still require an user input to confirm that current TK buffers will be saved (if not, proceed with renaming without saving buffers).
Like you, I also came to the conclusion that to do that, it should iterate over all buffers (or all unsaved ones to speed it up if possible), checking for filetype and saving the ones with `ft=telekasten`. But I need to look a bit more into the docs to do be able to do that properly.

Side note: maybe a function `save_all_tk_buffers()` could be useful to automatically save ZK buffers on exit if an option `save_on_exit = true` enables it?

Reply:
I agree and totally understand that one starts with the basic stuff - that's the most sensible approach anyway. 

I am so new to nvim-hacking and lua myself, unfortunately I don't know either how to check the filetype of a buffer - but I bet there's something useful in the help or you'll find something online really quickly.

Regarding the `save_all_tk_buffres()` - yeah, why not. Maybe we should define something like this (at the end of `telekasten.lua`:

```lua
M.utils = {}
M.utils.save_all_tk_buffers = your_awesome_function
```

That way we can collect all the neat function in `utils` and users can use them via `lua require('telekasten').utils.foo_bar()`